src/bot.coffee

```diff
@@ -173,7 +173,9 @@ class SlackBot extends Adapter
 
       when 'message', 'bot_message'
         @robot.logger.debug "Received message: '#{text}' in channel: #{channel.name}, from: #{user.name}"
-        @receive new TextMessage(user, text, message.ts)
+        textMessage = new TextMessage(user, text, message.ts)
+        textMessage.thread_ts = message.thread_ts
```

Comment:
Seems reasonable to me. Since `thread_ts` is a custom field specific to our message objects, it seems reasonable to apply it to the hubot message object explicitly like this, since it's not available in the scope of `TextMessage`.